homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -0,0 +1,349 @@
+        setpoint_cool = self.coordinator.get_coil_float(self._coil_setpoint_cool)
+
+        if mode == HVACMode.HEAT_COOL:
+            self._attr_target_temperature = None
+            self._attr_target_temperature_low = setpoint_heat
+            self._attr_target_temperature_high = setpoint_cool
+        elif mode == HVACMode.HEAT:
+            self._attr_target_temperature = setpoint_heat
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+        else:
+            self._attr_target_temperature = None
+            self._attr_target_temperature_low = None
+            self._attr_target_temperature_high = None
+
+        if prio := self.coordinator.get_coil_value(self._coil_prio):
+            if (
+                mixing_valve_state := self.coordinator.get_coil_value(
+                    self._coil_mixing_valve_state
+                )
+            ) and mixing_valve_state == 30:
+                self._attr_hvac_action = HVACAction.IDLE
+            elif prio == "HEAT":
+                self._attr_hvac_action = HVACAction.HEATING
+            elif prio == "COOLING":
+                self._attr_hvac_action = HVACAction.COOLING
+            else:
+                self._attr_hvac_action = HVACAction.IDLE
+        else:
+            self._attr_hvac_action = None
+
+        self.async_write_ha_state()
+
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        if not self.coordinator.last_update_success:
+            return False
+
+        if not self._coil_active_accessory_address:
+            return True
+
+        if active_accessory := self.coordinator.get_coil_value(
+            self._coil_active_accessory_address
+        ):
+            return active_accessory == "ON"
+
+        return False
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set target temperatures."""
+
+        hvac_mode = kwargs.get(ATTR_HVAC_MODE, self._attr_hvac_mode)
+
+        if (temperature := kwargs.get(ATTR_TEMPERATURE, None)) is not None:
+            if hvac_mode == HVACMode.HEAT:
+                await self.coordinator.async_write_coil(
+                    self._coil_setpoint_heat, temperature
+                )
+            elif hvac_mode == HVACMode.COOL:
+                await self.coordinator.async_write_coil(
+                    self._coil_setpoint_cool, temperature
+                )
+            else:
+                raise HomeAssistantError(
+                    f"Don't known which temperature to control for hvac mode: {self._attr_hvac_mode}"
+                )
+
+        if (temperature := kwargs.get(ATTR_TARGET_TEMP_LOW, None)) is not None:
+            await self.coordinator.async_write_coil(
+                self._coil_setpoint_heat, temperature
+            )
+
+        if (temperature := kwargs.get(ATTR_TARGET_TEMP_HIGH, None)) is not None:
```

Comment:
```suggestion
        if (temperature := kwargs.get(ATTR_TARGET_TEMP_LOW)) is not None:
            await self.coordinator.async_write_coil(
                self._coil_setpoint_heat, temperature
            )

        if (temperature := kwargs.get(ATTR_TARGET_TEMP_HIGH)) is not None:
```

`None` is the default for `get`